pennylane/compiler/compiler.py

```diff
@@ -20,16 +20,21 @@
 import pkg_resources
 
 
+class CompilerNotFoundError(RuntimeError):
+    """Exception raised when the compiler package is not installed."""
```

Comment:
I was also struggling about inheriting from `Exception` or `RuntimeError` :smile: 

The reason I ended up inheriting this compiler-not-being-found from `RuntimeError` is because it's a runtime issue that occurs when the interpreter is unable to locate or use the compiler, which is be definition a specific type of runtime error. However, it could also be represented as a more general exception that inherits from `Exception`. What do you think? Regarding Python recommended practices, I've come across both cases :thinking:

Reply:
I think it would be good to avoid the use of Python's `RuntimeErrror` for any Catalyst compilation errors, because it could be confusing to users. That's why in Catalyst we added `CompileError`.

Reply:
@dime10 What do you think of upstreaming `CompilerError` from Catalyst to PL?

Reply:
Yes you can probably just duplicate it in this module, it's very basic:
```
class CompileError(Exception):
    """Error encountered in the compilation phase."""
```
It's probably not worth it that either package has to import it from the other.